pkg/kv/export/export.go

```diff
@@ -0,0 +1,141 @@
+package export
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/treeverse/lakefs/pkg/kv"
+	"github.com/treeverse/lakefs/pkg/logging"
+)
+
+var (
+	ErrEmptyFile       = errors.New("empty file")
+	ErrBadHeader       = errors.New("bad header")
+	ErrBadEntry        = errors.New("bad entry")
+	ErrVersionMismatch = errors.New("version mismatch")
+)
+
+const (
+	exportPrefix = "lakefs_export_"
+)
+
+type Header struct {
+	Version   uint32
+	Timestamp time.Time
+}
+
+var (
+	firstHeader *Header
+	headerMu    sync.RWMutex
+)
+
+// TODO: (niro) implement export
+
+// TODO: (niro) Make private after migration
+func ImportFile(ctx context.Context, reader io.Reader, store kv.Store) error {
+	jd := json.NewDecoder(reader)
+	// Read header
+	if !jd.More() {
+		return ErrEmptyFile
+	}
+	var header Header
+	err := jd.Decode(&header)
+	// Decode does not return error on failure to Unmarshal
+	if err != nil || header == (Header{}) {
+		logging.Default().WithError(err).Error("Failed to decode header")
+		return ErrBadHeader
+	}
+	err = validateHeader(header)
+	if err != nil {
+		return err
+	}
+
+	var entry kv.Entry
+	for jd.More() {
+		err = jd.Decode(&entry)
+		// Decode does not return error on failure to Unmarshal
+		if err != nil || entry.Key == nil || entry.Value == nil {
+			logging.Default().WithError(err).Error("Failed to decode entry")
+			return ErrBadEntry
+		}
+		err = store.SetIf(ctx, entry.Key, entry.Value, nil)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func Import(ctx context.Context, store kv.Store, logger logging.Logger, path string) error {
+	reader, err := os.Open(path)
+	if err != nil {
+		return fmt.Errorf("error opening file (%s): %w", path, err)
+	}
+	defer reader.Close()
+
+	pattern := exportPrefix + strings.Split(filepath.Base(path), ".tar.gz")[0]
```

Comment:
Can the file not end with `.tar.gz`?

Reply:
And why expect a path if you're just using the last element of it?
e.g. passing `foo/bar/bla.tar.gz` is like passing `foo2/bar2/bla.tar.gz`?